uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/absence-tab/components/annual-entitlement-change-log.component.js

```diff
@@ -0,0 +1,319 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'common/moment',
+  'leave-absences/absence-tab/modules/components',
+  'common/models/contract'
+], function (_, moment, components) {
+  components.component('annualEntitlementChangeLog', {
+    bindings: {
+      contactId: '<',
+      periodId: '<',
+      dismissModal: '&'
+    },
+    templateUrl: ['settings', function (settings) {
+      return settings.pathTpl + 'components/annual-entitlement-change-log.html';
+    }],
+    controllerAs: 'changeLog',
+    controller: AnnualEntitlementChangeLog
+  });
+
+  AnnualEntitlementChangeLog.$inject = ['AbsencePeriod', 'AbsenceType',
+    'Entitlement', 'OptionGroup', 'shared-settings'];
+
+  function AnnualEntitlementChangeLog (AbsencePeriod, AbsenceType, Entitlement,
+  OptionGroup, sharedSettings) {
+    var entitlementsChangeLog = [];
+    var vm = this;
+
+    vm.absencePeriod = null;
+    vm.absenceTypes = [];
+    vm.changeLogRows = [];
+    vm.loading = { component: true };
+
+    (function init () {
+      loadAbsencePeriod()
+      .then(loadCalculationUnits)
+      .then(loadAbsenceTypes)
+      .then(loadChangeLog)
+      .then(appendCurrentEntitlementsToChangeLog)
+      .then(createChangeLogRows)
+      .then(removeRepeatedComments)
+      .then(highlightEntitlementWithComments)
+      .finally(function () {
+        vm.loading.component = false;
+      });
+    })();
+
+    /**
+     * Appends the current entitlements for the user and period into the change
+     * log's list. This is done because the logs don't return the current
+     * entitlement values, just the previous ones.
+     *
+     * @return {Promise}
```

Comment:
I think this function "returns appended current entitlements..." not just "appends". It would be also great to see what the Promise resolves with: `@return {Promise} resolves with all entitlements`

Reply:
This function returns a promise that resolves to undefined.